lib/pcb/svg-object-fns/create-svg-objects-from-pcb-note-dimension.ts

```diff
@@ -195,6 +190,49 @@ export function createSvgObjectsFromPcbNoteDimension(
     textAngle += 180
   }
 
+  // Apply text_ccw_rotation if provided (in degrees, subtract because SVG rotate is clockwise)
+  const finalTextAngle =
+    typeof text_ccw_rotation === "number" && Number.isFinite(text_ccw_rotation)
+      ? textAngle - text_ccw_rotation
+      : textAngle
+
+  // Calculate additional offset to prevent text from intersecting the line when rotated
+  let additionalOffset = 0
+  if (
+    text &&
+    typeof text_ccw_rotation === "number" &&
+    Number.isFinite(text_ccw_rotation)
+  ) {
+    // Estimate text dimensions (approximate for Arial font)
+    const textWidth = text.length * font_size * 0.6 // Approximate character width
+    const textHeight = font_size
+
+    // Calculate how much the rotated text extends toward the line
+    // The text is rotated by text_ccw_rotation relative to the line direction
+    const rotationRad = (text_ccw_rotation * Math.PI) / 180
+    const sinRot = Math.abs(Math.sin(rotationRad))
+    const cosRot = Math.abs(Math.cos(rotationRad))
+
+    // Half-width and half-height of text bounding box
+    const halfWidth = textWidth / 2
+    const halfHeight = textHeight / 2
+
+    // Maximum extension toward the line (perpendicular direction)
+    // This is the distance from text center to edge in the perpendicular direction
+    const maxExtension = halfWidth * sinRot + halfHeight * cosRot
+
+    // Add padding to ensure no intersection
+    additionalOffset = maxExtension + font_size * 0.3
+  }
+
+  const textOffset = arrow_size * 1.5 + additionalOffset
```

Comment:
```suggestion
  const textOffset = arrow_size * 1.5 + additionalOffset
```

1.5 seems like a magic number